assessmentModel/src/commonMain/kotlin/AssessmentExtensions.kt

```diff
@@ -39,21 +39,20 @@ private class FlatAnswersDefinitionGenerator(private val assessment: Assessment)
         val result = node.createResult()
         val pathSuffix = stepPath?.let { "${it}_" } ?: ""
         val identifier = result.identifier
-        var path : String? = "$pathSuffix$identifier"
-        if (node is Assessment && stepPath == null) {
-            // This is to prevent the assessment identifier from being included
-            path = null
-        }
+        val path : String? = if (node is Assessment && stepPath == null) null else "$pathSuffix$identifier"
         if (node is NodeContainer) {
             for(child in node.children) {
                 recursiveAdd(child, path)
             }
         }
 
         if (result is AnswerResult && result.answerType != null) {
+            var choices: Map<JsonPrimitive, String>? = null
+            if (node is ChoiceQuestion) {
+                choices = node.choices.associate { Pair(it.jsonValue(true)!!, it.label) }
```

Comment:
```suggestion
                choices = node.choices.mapNotNull {
                    val value = it.jsonValue(true)
                    if (it.selectorType == ChoiceSelectorType.Default && value != null && value !is JsonNull) {
                        Pair(value, it.label)
                    } else null
                }.toMap()
```

Don't want to map the "all" or "none" selectors and don't want to include the null.

The trick, really, is that AssessmentModelKMM was originally designed to be cross-platform with the limitation of Obj-c not supporting generics. The work-around was the oddly architected model that uses `JsonPrimitive` in the place of generics. Sorry.